drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -138,9 +159,7 @@ static int mcux_igpio_configure(const struct device *dev,
 
 #endif /* CONFIG_SOC_SERIES_IMX_RT10XX */
 
-	/* Init pin configuration struct, and use pinctrl api to apply settings */
-	assert(pin < config->mux_count);
-	memcpy(&pin_cfg.pinmux, &config->pin_muxes[pin], sizeof(pin_cfg));
+	memcpy(&pin_cfg.pinmux, &config->pin_muxes[cfg_idx], sizeof(pin_cfg));
```

Comment:
Should we retain the comment and the `assert `check?

Reply:
The assert was moved earlier in the function, because we were using `cfg_idx` to access the pin mux array before this assert was actually done. Now, the assertion will be checked before we use `cfg_idx`